### ISSUE TITLE
Illegal instruction

### ISSUE THREAD

**Issue by Licenser** [fede1024/rust-rdkafka#127]
There is a crash in rust-rdkafka that panics the program with an `Illegal instruction`.

It is not easy to reproduce but it seems to be related to Kafka disconnecting and connecting.

To reproduce I did:

* Set up Kafka with a topic (I set it to 32 partitions, not sure that affects the result)
* Set up a process that keeps sending data to the topic continously (~50k msg/s)
* Connect the simple consumer to the topic
* on the system running kafka execute: `while true; do sleep 30; systemctl stop kafka; sleep 30; systemctl start kafka; done` wait fro this to cycle through a few (can take a few hours :( )

**Comment by benesch** [comment]
This occurs when Kafka returns an error code that is as-yet unknown to librdkafka/rust-rdkafka. I suspect it's been resolved by the recent librdkafka bump, though is likely to happen again when Kafka introduces new error codes. Longer term we should move away from using `rustified_enum`: https://github.com/fede1024/rust-rdkafka/blob/6a3da62adad6e6116e48ff12093fc7f6311241e4/rdkafka-sys/build.rs#L68-L70.

**Comment by davidblewett** [comment]
> This occurs when Kafka returns an error code that is as-yet unknown to librdkafka/rust-rdkafka. I suspect it's been resolved by the recent librdkafka bump, though is likely to happen again when Kafka introduces new error codes. Longer term we should move away from using `rustified_enum`:
> 
> https://github.com/fede1024/rust-rdkafka/blob/6a3da62adad6e6116e48ff12093fc7f6311241e4/rdkafka-sys/build.rs#L68-L70
> 
> .

@Licenser Can you verify if a new build with the current version of `rust-rdkafka` resolves this panic? The greater question of whether or not to continue use `rustified_enum` remains.

**Comment by benesch** [comment]
> The greater question of whether or not to continue use `rustified_enum` remains.

Yeah, this would be a great starter project for someone. Ditching `rustified_enum` will make the API quite a bit less ergonomic, but the current situation means running rust-rdkafka against a new version of Kafka (or just a malicious Kafka broker) can result in undefined behavior, which is a bit terrifying.

**Comment by Licenser** [comment]
Hi @davidblewett - I really can't anymore the issue is quite old and we've been through a number of librdkafka updates since then and have resolved the issue (even so I don't remember any more how now) :(

**Comment by davidblewett** [comment]
Good to know. I'm going to keep this open until we come to a consensus on how to handle the enum issue.
